Indexing/TermCodeTree.cpp

```diff
@@ -64,39 +66,40 @@ void TermCodeTree::insert(TermInfo* ti)
     cctx.deinit(this);
   }
 
-  code.push(CodeOp::getSuccess(ti));
+  code.push(CodeOp::getSuccess(data));
   incorporate(code);  
   //@b incorporate should empty the code stack
   ASS(code.isEmpty());
 }
 
 //////////////// removal ////////////////////
 
-void TermCodeTree::remove(const TermInfo& ti)
+template<class Data>
+void TermCodeTree<Data>::remove(const Data& data)
 {
   static RemovingTermMatcher rtm;
   static Stack<CodeOp*> firstsInBlocks;
   firstsInBlocks.reset();
-  
-  FlatTerm* ft=FlatTerm::create(ti.term);
+
+  FlatTerm* ft=FlatTerm::create(data.term);
   rtm.init(ft, this, &firstsInBlocks);
   
-  TermInfo* rti;
+  Data* dptr;
   for(;;) {
     if (!rtm.next()) {
       ASSERTION_VIOLATION;
       INVALID_OPERATION("term being removed was not found");
     }
     ASS(rtm.op->isSuccess());
-    rti=rtm.op->getSuccessResult<TermInfo>();
-    if (*rti==ti) {
+    dptr=rtm.op->template getSuccessResult<Data>();
+    if (*dptr==data) {
       break;
     }
   }
   
   rtm.op->makeFail();
   
-  delete rti;
+  delete dptr;
```

Comment:
Could we initialise `dptr` to `nullptr`? This would prevent us deleting garbage if we fail to find something and allow us to `ASS(dptr)` here.